scipy/stats/tests/test_distributions.py

```diff
@@ -4315,6 +4318,99 @@ def test_burr_nan_mean_var_9544(self):
         assert_(np.isfinite(e4))
 
 
+class TestStudentizedRange:
+    # Critical Q values were selected from this table:
+    # https://link.springer.com/chapter/10.1007/978-1-4613-9629-1_12
+    # For alpha=.05 and .01, and for each value of v=[1, 3, 10, 20],
+    # a Q was picked from each table for k=[2, 8, 14, 20].
+    ks = [2, 8, 14, 20]
+    vs = [1, 3, 10, 20]
+    alpha05 = [17.97, 45.40, 54.33, 59.56,
+               4.501, 8.853, 10.35, 11.24,
+               3.151, 5.305, 6.028, 6.467,
+               2.950, 4.768, 5.357, 5.714]
+    alpha01 = [90.03, 227.2, 271.8, 298.0,
+               8.261, 15.64, 18.22, 19.77,
+               4.482, 6.875, 7.712, 8.226,
+               4.024, 5.839, 6.450, 6.823]
+    alphas = alpha05 + alpha01
+    ks_d_16 = np.ravel(list((repeat(ks, 4))))
+    vs_d_16 = np.ravel(list(zip(*(repeat(vs, 4)))))
```

Comment:
```suggestion
    vs_d_16 = sorted(vs * 4)
```

Reply:
But instead of generating these long lists yourself, use `itertools.product`:
```python3
>>> vs = [1, 3, 10, 20, 120, np.inf]
>>> ks = [2, 8, 14, 20]
>>> alpha = [0.05, 0.01]
>>> import itertools
>>> list(itertools.product(alpha, ks, vs))
[(0.05, 2, 1), (0.05, 2, 3), (0.05, 2, 10), (0.05, 2, 20), (0.05, 2, 120), (0.05, 2, inf), (0.05, 8, 1), (0.05, 8, 3), (0.05, 8, 10), (0.05, 8, 20), (0.05, 8, 120), (0.05, 8, inf), (0.05, 14, 1), (0.05, 14, 3), (0.05, 14, 10), (0.05, 14, 20), (0.05, 14, 120), (0.05, 14, inf), (0.05, 20, 1), (0.05, 20, 3), (0.05, 20, 10), (0.05, 20, 20), (0.05, 20, 120), (0.05, 20, inf), (0.01, 2, 1), (0.01, 2, 3), (0.01, 2, 10), (0.01, 2, 20), (0.01, 2, 120), (0.01, 2, inf), (0.01, 8, 1), (0.01, 8, 3), (0.01, 8, 10), (0.01, 8, 20), (0.01, 8, 120), (0.01, 8, inf), (0.01, 14, 1), (0.01, 14, 3), (0.01, 14, 10), (0.01, 14, 20), (0.01, 14, 120), (0.01, 14, inf), (0.01, 20, 1), (0.01, 20, 3), (0.01, 20, 10), (0.01, 20, 20), (0.01, 20, 120), (0.01, 20, inf)]
```

Reply:
This is very elegant, but but a concern I have is keeping the Q values (stores in `alphas`)  in the correct order as the output. I will play around with it a bit more to see how I can make it work.

Reply:
Or you could change the order of the product.